src/nifake/unit_tests/test_library_interpreter.py

```diff
@@ -410,6 +411,33 @@ def test_read_with_warning(self):
             assert test_error_desc in str(w[0].message)
             assert f'Warning {test_error_code} occurred.' in str(w[0].message)
 
+    def test_library_interpreter_always_uses_same_library_instance(self):
+        interpreter1 = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        interpreter2 = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        assert interpreter1 is not interpreter2
+        assert interpreter1._library is interpreter2._library
+
+    def test_set_runtime_environment_is_called_if_present(self):
+        nifake._library_interpreter._was_runtime_env_set = None
+        self.get_initialized_library_interpreter()
+        self.patched_library.niFake_SetRuntimeEnvironment.assert_called_once()
+
+    def test_set_runtime_environment_not_present_in_driver_runtime(self):
+        class TypesLibrary:
+            item = ""
+
+        nifake._library_interpreter._was_runtime_env_set = None
```

Comment:
Are you allowed to do this without using the "global" keyword, first?

Reply:
Hmm, you're not initializing the interpreter first, so it's probably still None.
Maybe you should just

```suggestion
        assert nifake._library_interpreter._was_runtime_env_set is None
```

Reply:
That will fail because the value is true at this point, because that part of the code was executed already. This ensures the set runtime environment function is called.

Reply:
Are you sure? I don't see us initilizing a LibraryInterpreter in `setup_method`, so how would the value have changed?